src/TokenParser/FromTokenParser.php

```diff
@@ -50,7 +51,7 @@ public function parse(Token $token): Node
 
         $stream->expect(Token::BLOCK_END_TYPE);
 
-        $var = new AssignNameExpression($this->parser->getVarName(), $token->getLine());
+        $var = new LocalVariable(null, $token->getLine());
```

Comment:
is it actually safe to migrate from a context variable to a local variable ? What happens for usages that are inside nested blocks, which won't have access to the same PHP local variables ?

Reply:
The `ImportNode` class only uses the `name` attribute from `$var` (the node is never compiled). So, here, we're only interested in the name, which is then going to be used as `$macros[NAME]`, which is propagated in nested blocks. See #4402 for a fix where we remove the wrong usage of the `AssignNameExpression` class.